cohortextractor/backends/base.py

```diff
@@ -43,6 +53,19 @@ class SQLTable:
     def learn_patient_join(self, source):
         raise NotImplementedError()
 
+    def _make_columns(self):
+        return [
+            self._make_column(name, column) for name, column in self._columns.items()
+        ]
+
+    def _make_column(self, name, column):
+        source = column.source or name
+        type_ = DEFAULT_SQLALCHEMY_TYPES[column.type]
```

Comment:
I've not been happy with any of my solutions to the problem of _really_ wanting to use a name that shadows something from the stdlib like this. Shall we go for underscore suffix as a general solution? It's fine by me.

Reply:
I'm just following SQLAlchemy here as it's the name it uses whenever you pass type as a keyword argument. I agree that none of the conventions are nice but following SQLAlchemy seems like a reasonable choice.